calyx/src/default_passes.rs

```diff
@@ -83,6 +83,7 @@ impl PassManager {
                 CollapseControl,
                 CompileRef, //Must run before 'resource-sharing'.
                 InferShare,
+                DeadCellRemoval,
```

Comment:
Why run this here? It is run as a part of `post-opt`. Does this make cell sharing faster?

Reply:
Some of the code depends on there being no dead cells, otherwise it will give an error. I can address this by pretty easily by just adding a couple if statements if you think that would be a better option.

Reply:
Update: I decided to get rid of the DeadCellRemoval pass before running CellShare and just make some minor edits to the code so that it will not give an error on dead cells.